.travis/travis.py

```diff
@@ -16,95 +48,199 @@ def test_diffs(diffs):
-    # TRAVIS_PULL_REQUEST_BRANCH = os.environ.get('TRAVIS_PULL_REQUEST_BRANCH','')
-    # TRAVIS_BUILD_DIR = os.environ.get('TRAVIS_BUILD_DIR',os.path.join(PWD,'..'))
-
-    REPO = os.environ['REPO']
-    TAG = os.environ['TAG']
-    OS_NAME = os.getenv('OS_NAME', '')
-    OS_CODE_NAME = os.getenv('OS_CODE_NAME', '')
-
-    TRAVIS_BRANCH = os.environ['TRAVIS_BRANCH']
-    TRAVIS_PULL_REQUEST_BRANCH = os.environ['TRAVIS_PULL_REQUEST_BRANCH']
-    TRAVIS_BUILD_DIR = os.environ['TRAVIS_BUILD_DIR']
-
-    print("REPO: ", REPO)
-    print("TAG: ", TAG)
-    print("OS_NAME: ", OS_NAME)
-    print("OS_CODE_NAME: ", OS_CODE_NAME)
-    print("TRAVIS_BRANCH: ", TRAVIS_BRANCH)
-    print("TRAVIS_PULL_REQUEST_BRANCH: ", TRAVIS_PULL_REQUEST_BRANCH)
-
-    # Expand the repo/tag directory
-    docker_repo_dir = os.path.join(TRAVIS_BUILD_DIR, REPO)
-    docker_tag_dir  = os.path.join(docker_repo_dir, TAG, OS_NAME, OS_CODE_NAME)
-
-    # Import the dockerfile generation script
-    spec = importlib.util.spec_from_file_location(
-        "create.dockerfiles",
-        os.path.join(docker_repo_dir, 'create_dockerfiles.py'))
-    create_dockerfiles = importlib.util.module_from_spec(spec)
-    spec.loader.exec_module(create_dockerfiles)
-
-    # Run the dockerfile generation script
-    create_dockerfiles.main(('dir', '-d' + docker_tag_dir))
-
-    # Create a git diff for the current repo
-    repo = git.Repo(TRAVIS_BUILD_DIR) #, odbt=git.GitCmdObjectDB)
+    # Public environment variables, available for pull requests from forks
+    HUB_REPO = os.environ['HUB_REPO']
+    HUB_RELEASE = os.environ['HUB_RELEASE']
+    HUB_OS_NAME = os.environ['HUB_OS_NAME']
+    HUB_OS_CODE_NAME = os.environ['HUB_OS_CODE_NAME']
+    GIT_BRANCH = os.environ['TRAVIS_BRANCH']
+    GIT_PULL_REQUEST_BRANCH = os.environ['TRAVIS_PULL_REQUEST_BRANCH']
+    GIT_UPSTREAM_REPO_SLUG = os.environ['TRAVIS_REPO_SLUG']
+    GIT_BUILD_DIR = os.environ['TRAVIS_BUILD_DIR']
+
+    print("HUB_REPO: ", HUB_REPO)
+    print("HUB_RELEASE: ", HUB_RELEASE)
+    print("HUB_OS_NAME: ", HUB_OS_NAME)
+    print("HUB_OS_CODE_NAME: ", HUB_OS_CODE_NAME)
+    print("GIT_UPSTREAM_REPO_SLUG: ", GIT_UPSTREAM_REPO_SLUG)
+    print("GIT_BRANCH: ", GIT_BRANCH)
+    print("GIT_PULL_REQUEST_BRANCH: ", GIT_PULL_REQUEST_BRANCH)
+
+    # Private environment variables, not available for pull requests from forks
+    GIT_USER = os.environ['GITHUB_USER']
+    GIT_EMAIL = os.environ['GITHUB_EMAIL']
+    GIT_TOKEN = os.environ['GITHUB_TOKEN']
+    GIT_AUTHOR = "{user} <{email}>".format(user=GIT_USER, email=GIT_EMAIL)
+    GIT_ORIGIN_REPO_SLUG = GIT_USER + '/' + GIT_UPSTREAM_REPO_SLUG.split('/')[1]
+    GIT_REMOTE_ORIGIN_TOKEN_URL = "https://{user}:{token}@github.com/{repo_slug}.git".format(
+        user=GIT_USER,
+        token=GIT_TOKEN,
+        repo_slug=GIT_ORIGIN_REPO_SLUG
+    )
+
+    # Initlise git interfaces
+    repo = git.Repo(GIT_BUILD_DIR)  # , odbt=git.GitCmdObjectDB)
```

Comment:
Is the goal to provide an object database in the future? or can this comment be removed?

Reply:
I wasn't sure if I should should use [GitCmdObjectDB](http://gitpython.readthedocs.io/en/stable/tutorial.html#gitcmdobjectdb) or stick with [GitDB](http://gitpython.readthedocs.io/en/stable/tutorial.html#gitdb). I'll try timing it, and use whatever is better.

Reply:
sounds good